google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -3343,6 +3628,28 @@ private void createSessions(final int sessionCount, boolean distributeOverChanne
     }
   }
 
+  class MultiplexedSessionConsumer implements SessionConsumer {
+
+    @Override
+    public void onSessionReady(SessionImpl session) {
+      final MultiplexedSession multiplexedSession = new MultiplexedSession(session);
+      synchronized (lock) {
+        multiplexedSessions.addFirst(multiplexedSession);
+        multiplexedSessionsInitialized.countDown();
+        numMultiplexedSessionsBeingCreated--;
+        while (!multiplexedSessionWaiters.isEmpty()) {
+          multiplexedSessionWaiters.poll().put(multiplexedSession);
+        }
+      }
+    }
+
+    @Override
+    public void onSessionCreateFailure(Throwable t, int createFailureForSessionCount) {
+      numMultiplexedSessionsBeingCreated--;
```

Comment:
This should also unblock any threads blocked in `waitOnMultiplexedSession` and propagate the error to those threads.